cartoframes/carto_vl/utils/html.py

```diff
@@ -0,0 +1,102 @@
+from warnings import warn
+from cartoframes import utils
+
+import os
+import json
+
+_HTML_TEMPLATE = '<iframe srcdoc="{srcdoc}" width="{width}" height="{height}"></iframe>'
+
+# CARTO VL
+_DEFAULT_CARTO_VL_PATH = 'https://libs.cartocdn.com/carto-vl/v1.1.1/carto-vl.min.js'
+
+# AIRSHIP
+_AIRSHIP_SCRIPT = '/packages/components/dist/airship.js'
+_AIRSHIP_BRIDGE_SCRIPT = '/packages/bridge/dist/asbridge.js'
+_AIRSHIP_STYLE = '/packages/styles/dist/airship.css'
+_AIRSHIP_ICONS_STYLE = '/packages/icons/dist/icons.css'
+
+_DEFAULT_AIRSHIP_COMPONENTS_PATH = 'https://libs.cartocdn.com/airship-components/v1.0.3/airship.js'
+_DEFAULT_AIRSHIP_BRIDGE_PATH = 'https://libs.cartocdn.com/airship-bridge/v1.0.3/asbridge.js'
+_DEFAULT_AIRSHIP_STYLES_PATH = 'https://libs.cartocdn.com/airship-style/v1.0.3/airship.css'
+_DEFAULT_AIRSHIP_ICONS_PATH = 'https://libs.cartocdn.com/airship-icons/v1.0.3/icons.css'
+
+
+class HTMLMap(object):
+    def __init__(self):
+        self.width = None
+        self.height = None
+        self.srcdoc = None
+
+    def set_content(
+        self, width, height, sources, bounds, creds=None, basemap=None,
+            _carto_vl_path=_DEFAULT_CARTO_VL_PATH, _airship_path=None):
+
+        html = self._parse_html_content(
+            sources, bounds, creds, basemap, _carto_vl_path, _airship_path)
+
+        self.width = width
+        self.height = height
+        self.srcdoc = utils.safe_quotes(html)
+
+    def _parse_html_content(
+        self, sources, bounds, creds=None, basemap=None,
+            _carto_vl_path=_DEFAULT_CARTO_VL_PATH, _airship_path=None):
+
+        html_template = os.path.join(
+            os.path.dirname(__file__), '..', '..', 'assets', 'vector.html')
+
+        token = ''
+
+        with open(html_template, 'r') as html_file:
+            srcdoc = html_file.read()
+
+        if creds is not None:
+            credentials = {
+                'username': creds.username(),
+                'api_key': creds.key(),
+                'base_url': creds.base_url()
+            }
+        else:
+            credentials = {
+                'username': 'cartovl',
```

Comment:
Maybe we should use `cartoframes` instead since there's a `cartoframes` examples account: https://cartoframes.readthedocs.io/en/stable/example_context.html